src/resources/Organizations/Organization.ts

```diff
@@ -9,11 +9,16 @@ import {
     OrganizationsStatusModel,
 } from './OrganizationInterfaces';
 
+type ListDependingOnPagination<T> = T extends ListOrganizationOptions
+    ? PageModel<OrganizationModel>
+    : OrganizationModel[];
 export default class Organization extends Resource {
     static baseUrl = '/rest/organizations';
 
-    list(options?: ListOrganizationOptions) {
-        return this.api.get<PageModel<OrganizationModel>>(this.buildPath(Organization.baseUrl, options));
+    list<PossiblyPaginated extends ListOrganizationOptions | null | undefined>(options?: PossiblyPaginated) {
```

Comment:
The type doesn't seem quite right now either, not sure I have a working solution to suggest though

![image](https://user-images.githubusercontent.com/35579930/112511218-98868500-8d68-11eb-9248-2a56f24af531.png)
⬆️   These should all be `OrganizationModel[]`

Reply:
> The type doesn't seem quite right now either, not sure I have a working solution to suggest though

Yeah, the `c.organization.list({})` is not handled correctly right now it seems, it's okay for the others though.

Reply:
@olamothe note that in the screenshot above the changes added by this PR are present, my point being this PR doesn't really solve the issue.

Reply:
Yes I know, in your example, if you remove the `this.list({})` it should be okay ? 

That's what I'm trying to explain, I'll see if I can better handle the "empty object" corner case.

Reply:
What about the `this.list()` case?

Reply:
Yes, it's a similar problem/same root cause as above.

Reply:
Verbose solution but I think it works:
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcCCATDBLG2IB2AhgDYDyUA5kQdgF5F6EBi2wJGcAvHAOQDOMRgFd+vOAB8+JbAGNgBfsHFTBUbAUoBuALAAofaEiw4GmMCgAzIvLgAZbIIrVaDJgTJh3-OAG99cIFwRFi4+MQkrOwY-AD8AFzooe6kzjT0jOFRHJJJOCnkVOluWWwcANoAurp6QXCW2CTmUAlwahraAUHQGBat7Zo1dWBElMCtBMIAtgBGFkNBKFAACqPjiZOz812B-NAwAEKI-TDqgzsIyOttpx01AL76huD7lyj2jjBprpmEnt4AOUIAOEJBIRBmJFQPAcTiKP3c-3CPgAZHAABQXABKwAAjsJsFBgBgADzLOQAaxJsK+8IyiK8yIANHwQvlwqRsjFeAA+HmSbF4glE0nk2RUmnfenhJGEfgs3gNJoWXn8iSC-GE4lkynUz5Skp-RlyhU9FV8gW1II4zUinXivVwlzSo3eBUjMaqy11G3C7ViiX6umGjzGxTuiyrT0W9VWwK+rWi3WS4O-UNuvh7WBHL2xn1CxP2wNO4pp2XhvhIFCq-QASieeirqGYEAg3D4llbvBqTbgByIUHbvBmA94Dd7NIAIsAUAQcJoyAQoxo0ySACr8nhruCgcxznwp50h8v8IEEEFgiFQi6xOAtiAXRL9qA1fSWYQEWTuOAyQRkiD8Pw2CQogy7EOYnC7goMQfCWCIymGPhSJMYK5B+vQNAQxI8uiECIa0ywAUBIFgYwxK1n4FxEjAwhQAQfDiEQPgfhSBAQAA7vRTGwTA06zvOlCLqR7j-oBwEkKBowrhBPIPA2ABuA4IMAggAIztr+MDor49y1j2KkwOpAD0RlwCOUAKUp5iCAATBpnzacEyQcpEZQxIk1mGXAun6bZcAmfUrZAA

Reply:
I went with overloading the function